consensus/state.go

```diff
@@ -1202,15 +1203,7 @@ func (cs *State) enterPrecommit(height int64, round int) {
 
 	// +2/3 prevoted nil. Unlock and precommit nil.
 	if len(blockID.Hash) == 0 {
-		if cs.LockedBlock == nil {
-			logger.Info("enterPrecommit: +2/3 prevoted for nil.")
-		} else {
-			logger.Info("enterPrecommit: +2/3 prevoted for nil. Unlocking")
-			cs.LockedRound = -1
-			cs.LockedBlock = nil
-			cs.LockedBlockParts = nil
-			cs.eventBus.PublishEventUnlock(cs.RoundStateEvent())
-		}
+		logger.Info("enterPrecommit: +2/3 prevoted for nil.")
```

Comment:
don't we need `if cs.LockedBlock == nil {` condition anyway?

Reply:
I don't think it makes a difference if we are locked onto something or not. If there is +2/3 votes nil then we vote nil